pkg/client/api.go

```diff
@@ -200,11 +239,21 @@ func (h *queryClient) Query(ctx context.Context, t model.Time, query Selector) (
 	}
 
 	var queryRes QueryResult
-	err = json.Unmarshal(res.Data, &queryRes)
-	return queryRes, err
+	if err := json.Unmarshal(res.Data, &queryRes); err != nil {
+		return queryRes, &Error{
+			Type:        ErrBadData,
+			ErrorMsg:    fmt.Sprintf("failed to unmarshal JSON response: %v", err),
+			StatusCode:  res.StatusCode, // Use the status code from the response
+			Query:       string(query),
+		}
+	}
+	// Attach the StatusCode from the APIResponse to the QueryResult
+	queryRes.StatusCode = res.StatusCode
```

Comment:
ditto: we should be assuming this is 200 OK and not setting this explicitly if there is no error